### DISCUSSION TITLE
Numpy Version?

### DISCUSSION THREAD

**Discussion by XaviMartiCamarasa** [online-ml/river#1092] [Q&A] [answered]
Hi I have had an issue and hat to install anaconda and also river, but I'm getting this error. I guess this is because the numpy version.
![image](https://user-images.githubusercontent.com/117338671/202431572-aa32c666-c5ca-4e7b-a8e9-809fd8bf4ca7.png)

At first sight it is correctly installed

![image](https://user-images.githubusercontent.com/117338671/202432131-6d0ce5b0-c210-4dc5-97ac-b61c08c1665a.png)
Any help would be welcome!

**Comment by MaxHalford** [comment, marked as answer]
This usually can be fixed by running ´pip install numpy --upgrade´. I thought this issue would be fixed with the latest version of River though 🤷‍♂️

**Reply by XaviMartiCamarasa** [reply]
I have already done it, but still not working
![image](https://user-images.githubusercontent.com/117338671/202434016-50cdc491-088a-4084-9bdc-33dbff282eb1.png)
I think  he is expecting a previous version, doesn't it?

**Reply by XaviMartiCamarasa** [reply]
For a strange reason, it is running now..... Many Thanks Max!

**Reply by Divya216** [reply]
still facing the issue when run,

from river import ensemble

ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 232 from C header, got 216 from PyObject

Already upgraded numpy

**Reply by Divya216** [reply]
I am working on google collab:
I have the below versions of numpy and river

Requirement already satisfied: river in /usr/local/lib/python3.8/dist-packages (0.14.0)
Requirement already satisfied: numpy>=1.23.4 in /usr/local/lib/python3.8/dist-packages (from river) (1.24.1)
 
i still get the below error when trying to import:
from river import metrics

ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 232 from C header, got 216 from PyObject

**Reply by Divya216** [reply]
Any suggestions?

**Reply by jomarca11** [reply]
Which python version are you using on google collab? You need 3.8 or above. 
But as far as I know collab is using 3.7.

**Reply by MaxHalford** [reply]
Colab moved to 3.8 back in November. I believe the issue comes from not restarting the notebook after installing River.

**Reply by Divya216** [reply]
restarted the environment. Works now :)  thank you!!

**Comment by XaviMartiCamarasa** [comment]
Have you tried to restart the environment and try again?

El mié., 4 ene. 2023 8:47, Divya216 ***@***.***> escribió:

> Any suggestions?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/online-ml/river/discussions/1092#discussioncomment-4589223>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A37HELZQRS2WN2H2GYECTJ3WQUTJBANCNFSM6AAAAAASDICXM4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 



INFORMACIÓ SOBRE PROTECCIÓ DE DADES DE LA UNIVERSITAT OBERTA DE 
CATALUNYA (UOC)

Us informem que les vostres dades identificatives i les 
contingudes en els missatges electrònics i fitxers adjunts es poden 
incorporar a les nostres bases de dades amb la finalitat de gestionar les 
relacions i comunicacions vinculades a la UOC, i que es poden conservar 
mentre es mantingui la relació. Si ho voleu, podeu exercir el dret a 
accedir a les vostres dades, rectificar-les i suprimir-les i altres drets 
reconeguts normativament adreçant-vos a l'adreça de correu emissora o a 
***@***.*** ***@***.***>.

Aquest missatge i qualsevol 
fitxer que porti adjunt, si escau, tenen el caràcter de confidencials i 
s'adrecen únicament a la persona o entitat a qui s'han enviat.

Així 
mateix, posem a la vostra disposició un delegat de protecció de dades que 
no només s'encarregarà de supervisar tots els tractaments de dades de la 
nostra entitat, sinó que us podrà atendre per a qualsevol qüestió 
relacionada amb el tractament de dades. La seva adreça de contacte és 
***@***.*** ***@***.***>.
INFORMACIÓN SOBRE PROTECCIÓN DE DATOS DE 
LA UNIVERSITAT OBERTA DE CATALUNYA (UOC)
Os informamos de que vuestros 
datos identificativos y los contenidos en los mensajes electrónicos y 
ficheros adjuntos pueden incorporarse a nuestras bases de datos con el fin 
de gestionar las relaciones y comunicaciones vinculadas a la UOC, y de que 
pueden conservarse mientras se mantenga la relación. Si lo deseáis, podéis 
ejercer el derecho a acceder a vuestros datos, rectificarlos y suprimirlos 
y otros derechos reconocidos normativamente dirigiéndoos a la dirección de 
correo emisora o a ***@***.*** ***@***.***>.
Este mensaje y 
cualquier fichero que lleve adjunto, si procede, tienen el carácter de 
confidenciales y se dirigen únicamente a la persona o entidad a quien se 
han enviado.
Así mismo, ponemos a vuestra disposición a un delegado de 
protección de datos que no solo se encargará de supervisar todos los 
tratamientos de datos de nuestra entidad, sino que podrá atenderos para 
cualquier cuestión relacionada con el tratamiento de datos. Su dirección de 
contacto es ***@***.*** ***@***.***>.


UNIVERSITAT OBERTA DE 
CATALUNYA (UOC) DATA PROTECTION INFORMATION
Your personal data and the data 
contained in your email messages and attached files may be stored in our 
databases for the purpose of maintaining relations and communications 
linked to the UOC, and the data may be stored for as long as these 
relations and communications are maintained. If you so wish, you can 
exercise your rights to access, rectification and erasure of your data, and 
any other legally held rights, by writing to the sender’s email address or 
***@***.*** ***@***.***>.
This message and, where 
applicable, any attachments are confidential and addressed solely to the 
individual or organization they were sent to.
The UOC has a data protection 
officer who not only supervises the data processing carried out at the 
University, but who will also respond to any questions you may have about 
this data processing. You can contact our data protection officer by 
writing to ***@***.*** ***@***.***>.

**Reply by Divya216** [reply]
restarted the environment. Works now :)  thank you!!

**Comment by jomarca11** [comment]
Agree with you

**Comment by albertfrancajosuacosta** [comment]
Hey.

I have same problem.

I use anaconda and spyder IDE or  jupyter and not work restart the environment or restart the computer.

**Reply by albertfrancajosuacosta** [reply]
I need to remove tensorflower, numpy, pandas, scipy.
After, i reinstall numpy, pandas, scipy and river.

Apparently solved in my case.

**Reply by jomarca11** [reply]
Good news! We have to be careful about library versions
